config/config_test.go

```diff
@@ -602,6 +602,19 @@ func TestUnmarshalSecretURL(t *testing.T) {
 	require.Equal(t, "http://example.com/se%20cret", u.String(), "SecretURL not properly unmarshaled in YAML.")
 }
 
+func TestHideSecretURL(t *testing.T) {
+	b := []byte(`"://wrongurl/"`)
+	var u SecretURL
+
+	err := json.Unmarshal(b, &u)
+	if err != nil {
+		errStr := err.Error()
+		if strings.Contains(errStr, "://wrongurl/") {
+			t.Fatal("config's String method reveals authentication credentials.")
+		}
+	}
```

Comment:
A bit shorter + ensuring that there's an error. 
```suggestion
	err := json.Unmarshal(b, &u)
	require.Error(t, err)
	require.NotContains(t, err.Error(), "wrongurl")
```